sycl/unittests/context_device/HasExtensionWordBoundary.cpp

```diff
@@ -0,0 +1,147 @@
+//==---- HasExtensionWordBoundary.cpp --- Test word boundary fix ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This test verifies that has_extension correctly matches full extension names
+// and doesn't match partial substrings.
+//
+//===----------------------------------------------------------------------===//
+
+#include <detail/device_impl.hpp>
+#include <gtest/gtest.h>
+#include <helpers/UrMock.hpp>
+#include <sycl/sycl.hpp>
+#include <ur_mock_helpers.hpp>
+
+using namespace sycl;
+
+static std::string MockExtensions = "";
```

Comment:
Nit; In case GTest ever decides to run cases in parallel, you could make this `thread_local` to avoid potential conflicts. I don't know if that can/will ever happen, but it's a simple way to make it more robust.

Reply:
I do believe that is the case, but I like to be safe rather than sorry. If it somehow starts using multi-threading, this will be a paint to debug.